src/response.rs

```diff
@@ -64,7 +122,58 @@ impl Response {
         }
     }
 
-    /// Get the `Headers`.
+    /// Get the `Headers` of this `Response`.
+    ///
+    /// ```rust
+    /// # extern crate reqwest;
+    /// # #[macro_use] extern crate error_chain;
+    /// # use std::fs;
+    /// # use std::io::{Read, Write};
+    /// # use reqwest::{Client, StatusCode};
+    /// # use reqwest::header::ContentType;
+    /// # use reqwest::mime::{Mime, TopLevel, SubLevel};
+    /// # error_chain! {
+    /// #     foreign_links {
+    /// #         Reqwest(reqwest::Error);
+    /// #         Io(::std::io::Error);
+    /// #     }
+    /// # }
+    /// # fn run() -> Result<()> {
+    /// let client = Client::new()?;
+    /// let mut resp = client.get("https://img.shields.io/crates/v/mime.png?label=mime")?.send()?;
+    /// match *resp.status() {
+    ///     StatusCode::Ok => (),
+    ///     status => bail!("Bad http response, status: {:?}", status),
+    /// };
+    ///
+    /// {
+    ///     let content_type = resp.headers().get::<ContentType>();
+    ///     if let Some(ct) = content_type {
+    ///         match **ct {
+    ///             Mime(TopLevel::Image, SubLevel::Png, _) => {
```

Comment:
Having just submitted a redesign of the mime crate, seeing it's current (v0.2) API still makes me sad. Checking the content-type is definitely something useful... but I do kind of wish we could keep that verbose mess out of here... (know I'm complaining about my own work, the mime crate, not your example).